science/iqtree/Portfile

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               github 1.0
+PortGroup               cmake 1.1
+
+name                    iqtree
+github.setup            Cibiv IQ-TREE 1.6.12
+revision                0
+checksums               rmd160  249414b567bc4f3d2a939d8f7e71cd72f653ed3c \
+                        sha256  9614092de7a157de82c9cc402b19cc8bfa0cb0ffc93b91817875c2b4bb46a284 \
+                        size    4864641
+
+categories              science
+platforms               darwin
+maintainers             nomaintainer
+license                 GPL-2
+
+description             Efficient phylogenomic software by maximum likelihood
+
+long_description        The IQ-TREE software was created as the successor of IQPNNI and \
+                        TREE-PUZZLE (thus the name IQ-TREE). IQ-TREE was motivated by the \
+                        rapid accumulation of phylogenomic data, leading to a need for \
+                        efficient phylogenomic software that can handle a large amount of \
+                        data and provide more complex models of sequence evolution. To this \
+                        end, IQ-TREE can utilize multicore computers and distributed parallel \
+                        computing to speed up the analysis. IQ-TREE automatically performs \
+                        checkpointing to resume an interrupted analysis.
+
+homepage                http://www.iqtree.org/
+github.tarball_from     archive
+github.master_sites     ${github.homepage}/archive/
+distname                v${version}
+
+depends_build-append    port:eigen3 \
+                        port:libomp
+
```

Comment:
also from the build log you will see that it uses MacPorts zlib library (if present), so to get reproducible builds you'll likely want to add a dependency to it. 

Along the same lines, the build script checks (at least if I remember correctly last time I looked at this while installing it on Linux) for the presence of `MPI`. Again for reproducibility, this should be disabled to make sure it doesn't get picked up by chance; if you want you can likely provide a variant to enable it.

Reply:
there appears to be some weird stuff going on with the zlib dependency, for example in the build log you see message like:
`ld: warning: object file (/opt/local/lib/libz.a(gzclose.o)) was built for newer OSX version (10.14) than being linked (10.7)`

It appears that the the CMakeLists.txt file they do:
```
# to be compatible back to Mac OS X 10.7
add_definitions("-mmacosx-version-min=10.7")
set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.7")
```

I am by no means an expert on this, but I'd suspect that this doesn't align well with the MacPorts way of building things. Perhaps @cjones051073 has some better insights here?
```

Reply:
yes, that could be problematic... 

If upstream don't provide an option to disable this 

a) Suggest to them to add an option, doing this unconditionally will interfere with build systems like MacPorts 
b) In the meantime, I would suggest using an `reinplace` to change it to the current SDK. something like

```
reinplace "s|mmacosx-version-min=10.7|mmacosx-version-min=${configure.sdk_version}|" ${worksrcpath}/CMakeLists.txt
```

Note that is untested, so make sure it works yourself...

Reply:
I just looked, an easier solution to me is to just patch (i.e., comment out/remove) everything except the last line from the CMakeLists.txt:
```
 elseif (APPLE)
        message("Target OS     : Mac OS X")
-       # to be compatible back to Mac OS X 10.7
-       if (IQTREE_FLAGS MATCHES "oldmac")
-               add_definitions("-mmacosx-version-min=10.5")
-               set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.5")
-       else()
-               add_definitions("-mmacosx-version-min=10.7")
-               set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.7")
-       endif()
+    ### to be compatible back to Mac OS X 10.7
+       ##if (IQTREE_FLAGS MATCHES "oldmac")
+       ##      add_definitions("-mmacosx-version-min=10.5")
+       ##      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.5")
+       ##else()
+       ##      add_definitions("-mmacosx-version-min=10.7")
+       ##      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.7")
+       ##endif()
     SET(CMAKE_FIND_LIBRARY_SUFFIXES .a ${CMAKE_FIND_LIBRARY_SUFFIXES})
```

at least that works fine for me.